R/htbatch.R

```diff
@@ -23,31 +23,38 @@ htbatch <- function (file, sigma.selection, ...) {
       rm (t)
 
 
+      if (is.na(qsardataframe$r1.ind.smiles[i]) == FALSE) {
 
-    } else {
+        t <- htdesc (smile = qsardataframe$r1.ind.smiles[i], HT.type = "induction", sigma.selection)
+        qsardataframe$r1.ind.sub.smiles[i] <- as.character (t$sub)
+        qsardataframe$r1.ind.mcs.index[i] <- t$tanimoto
+        qsardataframe$r1.ind.value[i] <- t$value
+        rm (t)
 
-      # if its not aromatic, than we will need to call htdesc
+      }
 
-      #converting SMIset into SDFSet required for evaluation in getfragmentsigma
 
-      # getting sigma value from htdescvalue function for r1taft
 
-      # calling htdesc to fill substitute mcs values
+    } else {
 
       # if the structure is aromatic, than we will set these to default values
 
-      # qsardataframe$r1.taft.smiles[i] <- "*C1=CC=CC=C1"
-      # qsardataframe$r1.es.smiles[i] <- "*C1=CC=CC=C1"
-      # qsardataframe$r1.ind.smiles[i] <- "*C1=CC=CC=C1"
+      qsardataframe$r1.taft.smiles[i] <- "*C1=CC=CC=C1"
+      qsardataframe$r1.ind.smiles[i] <- "*C1=CC=CC=C1"
 
       # calling htdesc to fill substitute mcs values
-
       t <- htdesc (smile = qsardataframe$r1.taft.smiles[i], HT.type = "taft", sigma.selection)
```

Comment:
idk if we want to implement it, but let me throw out a suggestion. htdesc calls are very expensive timewise, so it would be nice if we can reduce the number of htdesc calls in htbatch. If we know that  qsardataframe$r1.taft.smiles[i] is aromatic and set to *C1=CC=CC=C1, then we already know that qsardataframe$r1.taft.sub.smiles[i] and qsardataframe$r1.taft.mcs.index[i] are *C1=CC=CC=C1 and 1 respectively. Now, based on sigma.selection we also know the index for the qsardataframe$r1.taft.value[i] in the taft lookup table. 

So cant we directly just call that instead of htdesc? maybe write a small "helper" function to return the taft value for a specific index and sigma.selection parameter?

Reply:
This sounds good. We will create the helper function